sound/hda/hdac_device.c

```diff
@@ -566,7 +566,13 @@ EXPORT_SYMBOL_GPL(snd_hdac_get_connections);
  */
 int snd_hdac_power_up(struct hdac_device *codec)
 {
-	return pm_runtime_get_sync(&codec->dev);
+	int ret;
+
+	ret = pm_runtime_resume_and_get(&codec->dev);
+	if (ret < 0 && ret != -EACCES)
```

Comment:
This comment applies to other, similar patches:
- commit message have `-ENOACCES`, but it is `-EACCES`
- Why -EACCES is success and not error? It was treated as error before...
- `pm_runtime_resume_and_get()` return 0 on success or the error code on failure
```c
if (ret && ret != -EACCES)
	return ret;
```
or
```c
if (ret == -EACCES)
	ret = 0;

return ret;
```
might be better if -EACCES _is_ really not an error.

Reply:
-EACCES is returned by pm_runtime is disabled or not allowed. It's not an error. Ideally it should have been filtered at the source in the pm_runtime code.

if (ret && ret != -EACCES)
	return ret;

isn't very different from my suggestion

if (ret < 0 && ret != -EACCES)
	return ret;

I prefer to test for negative values explicitly.

Reply:
fixed the commit messages, see update

Reply:
Ok, but 'pm_runtime_resume_and_get()' will never return positive number, testing for 'ret < 0' gives the impression that a positive return number is possible and it is a success indicator.

Reply:
@ujfalusi this patch has been removed but even otherwise testing for ret < 0 makes sense. just ret != -EACCESS would include 0 as well

Reply:
@ranj063, oh, ok.
I was suggesting:
```c
if (ret && ret != -EACCES)
	return ret;

return 0;
```
or
```c
if (ret == -EACCES)
	ret = 0;

return ret;
```

Reply:
@ujfalusi there's already a precedent for using < 0 across drivers when using pm_runtime_get_sync, let's keep the same style as before....

Reply:
Right, it looks that way. Almost all `pm_runtime_resume_and_get()` is with `if (ret < 0)`.
In those cases the error check was not updated when converting to the simplified wrapper.

In case of `pm_runtime_get_sync()` it is valid as it can return positive number which is not an error, but the resume_and_get wrapper got it wrong (arguably).

Reply:
I stand corrected, `pm_runtime_resume_and_get()` does things correctly, it just wraps __pm_runtime_resume() and makes it return 0 or error code and removes the positive number as return value.

Eating up -EACCES does change the behavior as we did not done that in the past.

Reply:
@plbossart, `pm_runtime_get_sync()` is a wrapper for `__pm_runtime_resume()`, which is calling `rpm_resume()` which returns 0 on success, positive number if the status was already enabled.
pm_runtime_get_sync:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/power/runtime_pm.rst#n385
and pm_runtime_resume:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/power/runtime_pm.rst#n338

The `pm_runtime_resume_and_get()` is returning 0 if `retval >= 0` so the code does not need to check for positive numbers, `!ret` is success `ret != 0` is error.

Let me check the PR again.